Installer/source/main.cpp

```diff
@@ -825,32 +837,147 @@ s8 PatchTMD( u8 delete_mode )
 		}
 		SaveTmd++;
 	}
-	gprintf("checking Boot app SHA1 hash...");
-	gprintf("bootapp ( %d ) SHA1 hash = %02x%02x%02x%02x %02x%02x%02x%02x %02x%02x%02x%02x %02x%02x%02x%02x %02x%02x%02x%02x",rTMD->boot_index
-		,rTMD->contents[rTMD->boot_index].hash[0],rTMD->contents[rTMD->boot_index].hash[1],rTMD->contents[rTMD->boot_index].hash[2],rTMD->contents[rTMD->boot_index].hash[3]
-		,rTMD->contents[rTMD->boot_index].hash[4],rTMD->contents[rTMD->boot_index].hash[5],rTMD->contents[rTMD->boot_index].hash[6],rTMD->contents[rTMD->boot_index].hash[7]
-		,rTMD->contents[rTMD->boot_index].hash[8],rTMD->contents[rTMD->boot_index].hash[9],rTMD->contents[rTMD->boot_index].hash[10],rTMD->contents[rTMD->boot_index].hash[11]
-		,rTMD->contents[rTMD->boot_index].hash[12],rTMD->contents[rTMD->boot_index].hash[13],rTMD->contents[rTMD->boot_index].hash[14],rTMD->contents[rTMD->boot_index].hash[15]
-		,rTMD->contents[rTMD->boot_index].hash[16],rTMD->contents[rTMD->boot_index].hash[17],rTMD->contents[rTMD->boot_index].hash[18],rTMD->contents[rTMD->boot_index].hash[19]);
-	gprintf("generated priiloader SHA1 : ");
-	sha.Reset();
-	sha.Input(priiloader_app, priiloader_app_size);
-	if (!sha.Result(FileHash))
-	{
-		gprintf("could not compute Hash of Priiloader!");
-		r = -1;
-		goto _return;
-	}
-	if (!memcmp(rTMD->contents[rTMD->boot_index].hash, FileHash, sizeof(FileHash) ) )
+
+	if (is_vwii)
 	{
-		gprintf("no SHA hash change needed");
+		// First read the original app (which is now BC-NAND)
+		memset(status, 0, sizeof(fstats));
+		fd = ISFS_Open(original_app,ISFS_OPEN_READ);
+		if (fd < 0)
+		{
+			gprintf("Cannot open original app.");
+			goto _return;
+		}
+		if (ISFS_GetFileStats(fd,status) < 0)
+		{
+			gprintf("Cannot stat original app.");
+			r = -1;
+			goto _return;
+		}
+		AppData = (u8 *)memalign(32,ALIGN32(status->file_length));
+		if (AppData != NULL)
+			r = ISFS_Read(fd,AppData,status->file_length);
+		else
+		{
+			gprintf("Cannot allocate buffer for original app.");
+			r = -1;
+			goto _return;
+		}
+		ISFS_Close(fd);
+		fd = 0;
+		if (r < 0)

```

Comment:
maybe merge these 2 if statements? i hate nested if's haha

Reply:
I think I copied that code from above heh.
The `if (AppData)` seems redundant anyways, since that code isn't reached if the allocation failed.

Reply:
possible, at one point i went WAY WAY WAY overboard in the installer and its checks because of the version that could brick due to a HBC/libogc error.
if its redundant, maybe its better to remove it. 
in thema of leaving the code behind cleaner than how its found ^^;